bin/fetch_configlet.dart

```diff
@@ -0,0 +1,120 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:archive/archive.dart';
+
```

Comment:
Add a variable `var _operatingSystem = Platform.operatingSystem;`

Then on the first line of `main`, add `  if (_operatingSystem == 'macos') _operatingSystem = 'mac';`

That will fix an error when running on Mac. The mac zip/tar files do not have the `os` appended at the end.

However, there is a new issue that I don't know how to solve. View the output from the console below. I'll take a deeper dive later.

```
Downloading latest release from 
https://github.com/exercism/configlet/releases/download/v3.9.2/configlet-mac-64bit.tgz
Created: "bin/"

Created: "bin/a½xxî`ðôß&©÷çW-'è¯XÂðû!rxÓKKcîãð»¾àÍn¯^¼Ràý"

;i¾§tsÐ¸s÷\s½ {îayÅo1ÊUð{û­ÃÇ\3j{äÐ"

ê;öøØàWZÃaôlÕBÿ¼Cð¯	þáõ[ðoü[ÿÊþþbh~ØýXÙO6û1³ûÑ±cÐ2¡Â"

Failed with error:
FileSystemException: Cannot open file, path = 'bin/?$¼x6~|KjpJÇ	¯Mpë«vmÈ{wºz#1gàÖ÷7 PÎ{ÏÛ¬wwó6'»{z<W@(ü¡K Eºèé©±ý®C,l/·na9Þ\7ÞE:÷¢ËÔÕQ' (OS Error: No such file or directory, errno = 2)
```

Reply:
So maybe the contents of tar need different encoding to extract?. From the console output, it's creating two outputs from the archive. And when I downloaded the [Mac version](https://github.com/exercism/configlet/releases/download/v3.9.2/configlet-mac-64bit.tgz) it only has 1 file (configlet).

Reply:
@devkabiir I don't know. But I'll try to dig deeper into this today.

Reply:
I've updated the script to use `mac` instead of `macos`. Please try I again.

Reply:
Just tried it again:

```bash
dart bin/fetch_configlet.dart 
Created: "bin/"
Created: "bin/a½xx î`ðôß&©÷çW-'è¯XÂðû!r│ÓKK␌îãð»¾ àÍ┼ ¯^¼Ràý "
C⎼␊▒├␊␍: "␉␋┼/ÿû÷¿ ÿ¿þ

            ÷\⎽ ½ πî ▒≤Å⎺ 1ÊUðπû­ÃÇ\3┘ πäÐ" & 
ê;öø ØàWZÃ▒ô┌  ÕBÿ¼Cð¯?Æþáõ[ð⎺ü[ÿÊ þþ␉␤· Ø  ýXÙO6û1³ûÑ± ␌Ð 2¡Â"
F▒␋┌␊␍ ┬␋├␤ ␊⎼⎼⎺⎼:
F␋┌␊S≤⎽├␊└E│␌␊⎻├␋⎺┼: C▒┼┼⎺├ ⎺⎻␊┼ °␋┌␊, ⎻▒├␤ = '␉␋┼/?$¼│ 6·≠K┘⎻JÇ        ¯M⎻ë«┴ └Èπ ┬º≥#1±àÖ÷ 7 PÎπÏÛ¬┬┬ó6'»π≥  <W@
                                                                                                                  (ü¡K Eºè é©±ý®C,┌/·┼▒9Þ\7ÞE :÷¢ËÔÕ Q' (OS E⎼⎼⎺⎼: N⎺ ⎽┤␌␤ °␋┌␊ ⎺⎼ ␍␋⎼␊␌├⎺⎼≤, ␊⎼⎼┼⎺ = 2)
✘-1 ·/W⎺⎼┐⎽⎻▒␌␊/G␋├H┤␉/S├▒⎼±▒├⎺⎼/␍▒⎼├ [°␊├␌␤_␌⎺┼°␋±┌␊├.␍▒⎼├≠✚ 1…9⚑ 8]
```
Not the last line with the ✘, that's how my bash displays my current folder. Before it would display it like:
`✘-254 ~/Workspace/GitHub/Stargator/dart [fetch_configlet.dart|…5⚑ 8]`

So whatever happened changed how my terminal displays text.

Closing the terminal tab and opening a new one, resolve the issue. But that's still odd. I don't recommend printing out the contents of files, if that is what is happening.

Reply:
This is definitely caused by me printing out the file names. Although they should not cause any escaping or un-escaping. This looks to me like an encoding/decoding problem when I try to un-archive the contents. This leads to having wrong/weird file names, which in turn break your terminal. I will take another look at it soon